front-end/src/hooks/useReadTodo.ts

```diff
@@ -0,0 +1,20 @@
+import {useEffect} from "react";
+import { TodoAPI } from "../services";
+import { useAppDispatch, useAppSelector } from "../app/hooks";
+import { RootState } from "../app/store";
+import { readTodoAction } from "../features/todo/todoSlice";
+
+
+export default function useReadTodo() {
+    const todo = useAppSelector((state: RootState) => state.todo);
+    const dispatch = useAppDispatch();
+    useEffect( () => {
+        TodoAPI.readTodo()
+        .then((response) => {
+            dispatch(readTodoAction(response));
+        }).catch((error) => {
+            console.log(error);
```

Comment:
Handle error state, by setting `todoList` to `null` and then rendering the error on the UI if `todoList` is `null`.